docs/start_guide.md

```diff
@@ -0,0 +1,257 @@
+        - file://artifacts/kubernetes/hello-apache-pod.json
+      marathon:
+        - file://artifacts/marathon/helloapache.json
+```
+
+__Graph__ is the most important section. In here we will indicate all the default parameters as well as all associated artifacts.
+
+```yaml
+- name: image
+  description: The webserver image
+  default: centos/httpd
+```
+There will likely be many parameters that need to be exposed at deployment. It's best to provide defaults whenever possible. Variable templating is used within artifact files. For example: `$image` within `artifacts/kubernetes/hello-apache-pod.json` becomes `centos/httpd`.
+
+Not providing a default variable will require Atomic App to ask the user. Alternatively, an `answers.conf` file can be provided.
+
+```yaml
+artifacts:
+  docker:
+    - file://artifacts/docker/hello-apache-pod_run
+  kubernetes:
+    - file://artifacts/kubernetes/hello-apache-pod.json
+  marathon:
+    - file://artifacts/marathon/helloapache.json
+```
+In order to use a particular provider, a name and file reference(s) are required.
+
+```yaml
+kubernetes:
+  - file://artifacts/kubernetes/hello-apache-pod.json
+  - file://artifacts/kubernetes/hello-apache-service.json
+  - file://artifacts/kubernetes/hello-apache-rc.json
+```
+For example, multiple files can be specified for the `kubernetes` provider.
+
+### ~/helloapache/artifacts/docker/hello-apache-pod_run
+```sh
+docker run -d -p $hostport:80 $image
+```
+Each artifact uses variable replacement values. For our Docker provider, we substitute the port number with `$hostport` as indicated by our graph in our `Nulecule` file. The same as our `$image` variable.
+
+### ~/helloapache/artifacts/kubernetes/hello-apache-pod.json
+```json
+"image": "$image",
+"name": "helloapache",
+"ports": [
+    {
+        "containerPort": 80,
+        "hostPort": $hostport,
+        "protocol": "TCP"
+    }
+```
+
+Similarly, the kubernetes provider uses both `$image` and `$hostport` variables for pod deployment.
+
+### ~/helloapache/answers.conf.gen
+
+`answers.conf.gen` is an answers file created when fetching. It is a generated ini file that provides parameter answers for non-interactive deployments.
+
+```ini
+[helloapache-app]
+image = centos/httpd
+hostport = 80
+
+[general]
+namespace = default
+provider = kubernetes
+```
+
+Default values such as the provider as well as the namespace can be provided.
+
+In order to use an answers file, simply specify the location of the file when deploying:
+```sh
+atomicapp run -a answers.conf.sample .
+```
+
+### Conclusion
+
+TODO: add more info here, not done yet. need to go more into detail on deployment after you've build the container.
+
+Congrats! We've just gone through the entirety of an example container.
+
+Now you can build and run your first app!
+
+```sh
+docker build -t myatomicapp .
+sudo atomicapp run myatomicapp
+```
+
+will continue later point, cdrage feb 26 16:59 est
+
+# More quickstart examples
+
+__GitLab on Kubernetes:__
```

Comment:
Yeah... until we use the "official" containers, gitlab would be a bad example, I'll add something else here (if any)